src/sst/core/timeConverter.h

```diff
@@ -30,6 +30,12 @@ class TimeConverter
     friend class TimeLord;
 
 public:
+    /**
+       Create a new TimeConverter object from a TimeConverter*
+       @param tc TimeConverter to initialize factor from
+     */
+    TimeConverter(TimeConverter* tc) : factor(tc->getFactor()) {}
```

Comment:
You could just initialize with tc->factor since they are the same class and can access private data members.